syntax = "proto3";

package omogenexec.proto;

import "api/execspec.proto";

message ContainerExecution {
  // The command to execute.
  omogenexec.api.Command command = 1;

  // The environment to execute the command in.
  omogenexec.api.Environment environment = 2;

  // The process limit to apply.
  int32 process_limit = 3;
}

message ContainerTermination {
  // An error encountered by the sandbox when trying to run the program.
  message Error {
    string error_message = 1;
  }

  oneof termination {
    // If the process terminated by a signal, it is set here.
    omogenexec.api.Termination.Signal signal = 1;
    // If the process terminated normally, the exit code is set here.
    omogenexec.api.Termination.Exit exit = 2;
    // If the sandbox encountered an error while trying to run the program, it
    // is set here.
    Error error = 3;
  }
}
